File: java/src/main/java/br/com/rh/models/PersonalData.java
package br.com.rh.models;

import br.com.rh.enums.Role;

import java.math.BigDecimal;

public class PersonalData {
    private String name;
    private String cpf;
    private Role role;
    private BigDecimal salary;

    public PersonalData(String name, String cpf, Role role, BigDecimal salary) {
        this.name = name;
        this.cpf = cpf;
        this.role = role;
        this.salary = salary;
    }

    public String getName() {
        return name;
    }

    public String getCpf() {
        return cpf;
    }

    public Role getRole() {
        return role;
    }

    public void setRole(Role role) {
        this.role = role;
    }

    public BigDecimal getSalary() {
        return salary;
    }

    public void setSalary(BigDecimal salary) {
        this.salary = salary;
    }
}

File: typescript/src/models/personal-data.ts
import { Role } from "../enums/role.enum";

export class PersonalData {
  private _name: string;
  private _cpf: string;
  private _role: Role;
  private _salary: number;

  constructor(name: string, cpf: string, role: Role, salary: number) {
    this._name = name;
    this._cpf = cpf;
    this._role = role;
    this._salary = salary;
  }

  get name() {
    return this._name;
  }

  get cpf() {
    return this._cpf;
  }

  get role() {
    return this._role;
  }

  get salary() {
    return this._salary;
  }

  set role(role: Role) {
    this._role = role;
  }

  set salary(salary: number) {
    this._salary = salary;
  }
}
